### PR TITLE
ref(ui) Remove more usage of createReactClass

### PR DESCRIPTION
**Pull request by markstory**
Remove a few more usages of createReactClass()

### CODE REVIEW

File: src/sentry/static/sentry/app/components/group/releaseChart.jsx [hunk 1]
```diff
@@ -21,15 +18,16 @@ const GroupReleaseChart = createReactClass({
     firstSeen: PropTypes.string,
     lastSeen: PropTypes.string,
     title: PropTypes.string,
-  },
+  };
 
-  getInitialState() {
-    return this.getNextState(this.props);
-  },
+  constructor(props) {
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/group/releaseChart.jsx]: I think minorly more correct would be `...args` and `super(...args)`.

File: src/sentry/static/sentry/app/components/group/releaseChart.jsx [hunk 2]
```diff
@@ -165,13 +163,13 @@ const GroupReleaseChart = createReactClass({
           label={t('events')}
           markers={markers}
           barClasses={['release', 'environment', 'inactive']}
-          tooltip={this.renderTooltip}
+          tooltip={this.renderTooltip.bind(this)}
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/group/releaseChart.jsx]: I would prefer making `renderTooltip` a class property, that way we don't rebind and create a new function on every render.

File: src/sentry/static/sentry/app/components/group/releaseChart.jsx [hunk 3]
```diff
@@ -21,15 +18,16 @@ const GroupReleaseChart = createReactClass({
...
     title: PropTypes.string,
-  },
+  };
 
-  getInitialState() {
-    return this.getNextState(this.props);
-  },
+  constructor(...args) {
+    super(...args);
+    this.state = this.getNextState(args[0]);
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/group/releaseChart.jsx]: Should we define the param name as props?

alternatively, class props should work too

`
state = this.getNextState(this.props);
`

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.jsx [hunk 4]
```diff
@@ -17,35 +16,31 @@ import SentryTypes from 'app/sentryTypes';
...
 
   componentDidUpdate(prevProps) {
     const queryHasChanged = !isEqual(prevProps.query, this.props.query);
     if (queryHasChanged || prevProps.params.tagKey !== this.props.params.tagKey) {
       this.fetchData();
     }
-  },
+  }
 
   async fetchData() {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.jsx]: Does `this` need to be bound?

Comment (markstory) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.jsx]: I don't think so? This method is called from inside componentDidUpdate. Does the execution scope change after the `await` resolves?

Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.jsx]: Get's called here too https://github.com/getsentry/sentry/pull/14427/files#diff-d0c0e2ebfb90779910356bf000df25b4R94

Comment (markstory) [reply src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.jsx]: Missed that. Thanks :+1:

File: src/sentry/static/sentry/app/components/group/releaseChart.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/group/releaseChart.jsx]: React docs always use `props` fwiw

**Review by billyvg — APPROVED** [review]

**Pull request merged**